ibm/config.go

```diff
@@ -594,91 +594,368 @@ func (sess clientSession) IAMNamespaceAPI() (*ns.IbmCloudFunctionsNamespaceAPIV1
+			ZoneIdentifier: core.StringPtr(""),
+			Authenticator:  sess.cisSSLClient.Service.Options.Authenticator,
+		}
+
+		cisSSLClient, cisSSLErr := cissslv1.NewSslCertificateApiV1(cisSSLOpt)
+		if cisSSLErr != nil {
+			return nil, fmt.Errorf(
+				"Error occured while configuring CIS SSL certificate client: %s",
+				cisSSLErr)
+		}
+		return cisSSLClient, nil
+	}
 	return sess.cisSSLClient, sess.cisSSLErr
 }
 
 // CIS WAF Packages
 func (sess clientSession) CisWAFPackageClientSession() (*ciswafpackagev1.WafRulePackagesApiV1, error) {
+	if sess.cisWAFPackageErr == nil {
+		cisWAFPackageOpt := &ciswafpackagev1.WafRulePackagesApiV1Options{
+			URL:           sess.cisWAFPackageClient.Service.Options.URL,
+			Crn:           core.StringPtr(""),
+			ZoneID:        core.StringPtr(""),
+			Authenticator: sess.cisWAFPackageClient.Service.Options.Authenticator,
+		}
+		cisWAFPackageClient, cisWAFPackageErr :=
+			ciswafpackagev1.NewWafRulePackagesApiV1(cisWAFPackageOpt)
+		if cisWAFPackageErr != nil {
+			return nil, fmt.Errorf(
+				"Error occured while configuration CIS WAF Package client: %s",
+				cisWAFPackageErr)
+		}
+		return cisWAFPackageClient, nil
+	}
 	return sess.cisWAFPackageClient, sess.cisWAFPackageErr
 }
 
 // CIS Zone Settings
 func (sess clientSession) CisDomainSettingsClientSession() (*cisdomainsettingsv1.ZonesSettingsV1, error) {
+	if sess.cisDomainSettingsErr == nil {
+		cisDomainSettingsOpt := &cisdomainsettingsv1.ZonesSettingsV1Options{
+			URL:            sess.cisDomainSettingsClient.Service.Options.URL,
+			Crn:            core.StringPtr(""),
+			ZoneIdentifier: core.StringPtr(""),
+			Authenticator:  sess.cisDomainSettingsClient.Service.Options.Authenticator,
+		}
+		cisDomainSettingsClient, cisDomainSettingsErr :=
+			cisdomainsettingsv1.NewZonesSettingsV1(cisDomainSettingsOpt)
+		if cisDomainSettingsErr != nil {
+			return nil, fmt.Errorf(
+				"Error occured while configuring CIS Domain Settings client: %s",
+				cisDomainSettingsErr)
+		}
+		return cisDomainSettingsClient, nil
+	}
 	return sess.cisDomainSettingsClient, sess.cisDomainSettingsErr
 }
 
 // CIS Routing
 func (sess clientSession) CisRoutingClientSession() (*cisroutingv1.RoutingV1, error) {
+	if sess.cisRoutingErr == nil {
+		cisRoutingOpt := &cisroutingv1.RoutingV1Options{
+			URL:            sess.cisRoutingClient.Service.Options.URL,
+			Crn:            core.StringPtr(""),
+			ZoneIdentifier: core.StringPtr(""),
+			Authenticator:  sess.cisRoutingClient.Service.Options.Authenticator,
+		}
+		cisRoutingClient, cisRoutingErr :=
+			cisroutingv1.NewRoutingV1(cisRoutingOpt)
+		if cisRoutingErr != nil {
+			return nil, fmt.Errorf(
+				"Error occured while configuring CIS Routing client: %s",
+				cisRoutingErr)
+		}
+		return cisRoutingClient, nil
+	}
 	return sess.cisRoutingClient, sess.cisRoutingErr
 }
 
 // CIS Cache service
 func (sess clientSession) CisCacheClientSession() (*ciscachev1.CachingApiV1, error) {
+	if sess.cisCacheErr == nil {
```

Comment:
Same here:

```go
if sess.cisCacheErr != nil {
	return sess.cisCacheClient, sess.cisCacheErr
}

cisCacheOpt := &ciscachev1.CachingApiV1Options{
	URL:           sess.cisCacheClient.Service.Options.URL,
	Crn:           core.StringPtr(""),
	ZoneID:        core.StringPtr(""),
	Authenticator: sess.cisCacheClient.Service.Options.Authenticator,
}
cisCacheClient, cisCacheErr :=
	ciscachev1.NewCachingApiV1(cisCacheOpt)
if cisCacheErr != nil {
	return nil, fmt.Errorf(
		"Error occured while configuring CIS Caching client: %s",
		cisCacheErr)
}

return cisCacheClient, nil

```

Reply:
I have upgraded sdk and regenerated cis sdk. Modified the code as per your suggession.